pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -50,13 +50,24 @@ func New(config *api.Config) (*OnBuild, error) {
 		tar:    tar.New(),
 	}
 	// Use STI Prepare() and download the 'run' script optionally.
-	s, err := sti.New(config)
+	s, err := sti.New(config, overrides)
 	s.SetScripts([]string{}, []string{api.Assemble, api.Run})
-	downloader, sourceUrl, err := scm.DownloaderForSource(config.Source)
-	if err != nil {
-		return nil, err
+
+	downloader := overrides.Downloader
+	if downloader == nil {
+		d, sourceURL, err := scm.DownloaderForSource(config.Source)
+		if err != nil {
+			return nil, err
+		}
+		downloader = d
+		config.Source = sourceURL
+	}
+
+	b.source = onBuildSourceHandler{
```

Comment:
Fixed it for you - including saving you from a govet error for anonymous
field declarations

On Mon, Oct 19, 2015 at 5:48 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/strategies/onbuild/onbuild.go
> <https://github.com/openshift/source-to-image/pull/310#discussion_r42430517>
> :
>
> >  	s.SetScripts([]string{}, []string{api.Assemble, api.Run})
> > -	downloader, sourceUrl, err := scm.DownloaderForSource(config.Source)
> > -	if err != nil {
> > -		return nil, err
> > +
> > +	downloader := overrides.Downloader
> > +	if downloader == nil {
> > +		d, sourceURL, err := scm.DownloaderForSource(config.Source)
> > +		if err != nil {
> > +			return nil, err
> > +		}
> > +		downloader = d
> > +		config.Source = sourceURL
> > +	}
> > +
> > +	b.source = onBuildSourceHandler{
>
> i liked the old line 61 better than your new one. :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/310/files#r42430517>.
>